src/contexts/WalletContext/index.js

```diff
@@ -43,8 +43,25 @@ const useEvmWallet = pool => {
   }, [refetch]);
 
   const callContract = useCallback(async (address, abi, method, params = [], isSend = false) => {
-    const contract = new ethers.Contract(address, abi, isSend ? signer : provider);
-    return contract[method](...params);
+    if (isSend) {
+      const contract = new ethers.Contract(address, abi, signer);
+      return contract[method](...params);
+    }
+
+    async function call(index = 0) {
+      if (index >= provider.providerConfigs.length) {
+        throw new Error('Error calling contract');
+      }
+      try {
+        const providerConfigs = [...provider.providerConfigs].sort((a, b) => a.priority - b.priority);
+        const contract = new ethers.Contract(address, abi, providerConfigs[index].provider);
```

Comment:
Sorting and getting provider by index may play a cruel joke. Let's assume priority of the first provider was decreased for any reasons and after sorting it becomes a second one.  This way after index incrementing you will select the same provider. I suggest to remove providers sorting by priority here, wdyt?

Reply:
I'm not quite clear on the problem. How will the priority change during the call? The priority is set in the config and can only be updated with a page reload.